src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -66,14 +74,116 @@ class EventsResource extends Resource {
  * GET all events
  */
     @GET
-    @Produces(MediaType.APPLICATION_JSON)
+    @Produces([MediaType.APPLICATION_JSON, "text/csv"])
+    //TODO Add CSV MediaType to response object
     public Response getEvents(@Auth AuthenticatedUser _,
                               @QueryParam('format') String format) {
-        def events = eventsDAO.getEvents()
-//@TODO add methods for returning csv and ics formats
-//        else if (format == "csv") {
-//
-//        } else if (format == "ics") {
+        List<ResourceObject> events = eventsDAO.getEvents()
+        def resultObject
+
+        if (!format) {
+            events.each {
+                resultObject.data += new ResourceObject(
+                        id: it.eventID,
+                        type: 'event',
+                        attributes: it
+                )
+            }
+        } else if (format == "csv") {
```

Comment:
This is too much code for this one method. Could you move all CSV manipulation code to CSVHelperFunctions.groovy? It could be split up into several static methods (public or private as needed), then for this EventsResource, it could just call one method to get a CSV. All the csv method(s) should need is a ResultObject with all events and instances.

```
} else if (format == "csv") {
    def allEventsCsv = CSVHelperFunctions.getCSV(allEvents)
    return Response.ok(allEventsCsv, "text/csv").build()
}